tests/js/client/aql/aql-optimizer-push-limit-into-index.js

```diff
@@ -0,0 +1,187 @@
+        "FOR i IN " + c.name() + " FILTER i.license IN ['cc-by-sa', 'cc-by-nc', 'foo'] and i.foo == true SORT i.date_created LIMIT 100 RETURN i._key",
+      ];
+ 
+      for (var i = 0; i < queries.length; ++i) {
+        let plan = db._createStatement(queries[i]).explain().plan;
+        let indexNode = plan.nodes[1];
+
+        assertEqual("IndexNode", indexNode.type);
+        assertEqual(indexNode.limit, 0);
+        assertEqual(-1, plan.rules.indexOf("push-limit-into-index"));
+      }
+    },
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test case with additional OR conditional filtering in index node
+////////////////////////////////////////////////////////////////////////////////
+    testPushLimitIntoIndexRuleORConditionalFiltering : function () {
+      var query = "FOR i IN " + c.name() + " FILTER i.license IN ['cc-by-sa', 'cc-by-nc', 'foo'] OR i.date_created > 500 SORT i.date_created LIMIT 100 RETURN i._key";
+
+      let plan = db._createStatement(query).explain().plan;
+      let indexNodes = plan.nodes.filter(function(n) { return n.type === 'IndexNode'; });
+
+      assertEqual(0, indexNodes.length);
+      assertEqual(-1, plan.rules.indexOf("push-limit-into-index"));
+    },
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test case with double loop
+////////////////////////////////////////////////////////////////////////////////
+    testPushLimitIntoIndexRuleDoubleLoop : function () {
+      var query = "FOR i in 0..2 FOR j IN " + c.name() + " FILTER j.license IN ['cc-by-sa', 'cc-by-nc', 'foo'] SORT j.date_created LIMIT 100 RETURN j._key";
+
+      let plan = db._createStatement(query).explain().plan;
+      let indexNode = plan.nodes[3];
+
+      assertEqual("IndexNode", indexNode.type);
+      assertEqual(indexNode.limit, 0);
+      assertEqual(-1, plan.rules.indexOf("push-limit-into-index"));
+    },
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test case with offset
+////////////////////////////////////////////////////////////////////////////////
+    testPushLimitIntoIndexRuleWithOffset : function () {
+      var query = "FOR i IN " + c.name() + " FILTER i.license IN ['cc-by-sa', 'cc-by-nc', 'foo'] SORT i.date_created LIMIT 8, 100 RETURN i._key";
+
+      let plan = db._createStatement(query).explain().plan;
+      let indexNode = plan.nodes[1];
+
+      assertEqual("IndexNode", indexNode.type);
+      assertEqual(indexNode.limit, 108);
+      assertNotEqual(-1, plan.rules.indexOf("push-limit-into-index"));
+    },
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test case when rule does not trigger
+////////////////////////////////////////////////////////////////////////////////
+    testPushLimitIntoIndexRuleDoesNotTrigger: function () {
```

Comment:
It would be good to also add tests with other indexes, e.g. single attribute indexes, compound indexes with nested attributes names, indexes with 3 attributes etc.